lib/js/composables/hover-state.ts

```diff
@@ -0,0 +1,38 @@
+import { Ref, ref } from 'vue';
+import { isTouchDevice } from '../utils/device';
+
+interface HoverState {
+	hovered: Ref<boolean>;
+	mouseOver: () => void;
+	mouseLeave: () => void;
+	touchStart: () => void;
+	touchEnd: () => void;
+}
+
+export function useHoverState(): HoverState {
+	const hovered = ref(false);
+
+	return {
+		hovered,
+		mouseOver() {
+			if (!isTouchDevice()) {
```

Comment:
We can cache the result of this function, I don't think it's possible to change in runtime.